### ISSUE TITLE
main() function

### ISSUE THREAD

**Issue by yoshw** [larryhudson/lectureDL#2]
Great work, Larry! I can see I'm not going to have too much to teach you this semester. :)

One improvement you could make to lectureDL.py is to move all the top-level statements (i.e. everything from line 44 onwards) into a `main()` function. Then add the following code (which is standard Python boilerplate) at the bottom of your file:

```
if __name__ == "__main__":
    main()
```

This allows Python to check whether the user is running your code from the command line (in which case `main()` will execute) or merely importing your code (using `import lectureDL`), in which case they probably DON'T want all the `main()` code to execute.

It also makes it cleaner & easier when you add command line arguments to your program, since you can add simple command-line argument checks/logic inside the `if __name__ == "__main__"` block to, for instance, stop `main()` from executing if incorrect arguments are provided.

**Comment by larryhudson** [comment]
Thanks for the feedback Yosh! I'll play around with the command-line arguments, not sure if I completely understand what they do at this stage so I'll do a bit of reading. I'm going to add a way to skip the `input` questions inside the script so that it will just run based on the arguments passed when the script is run, for example something like `python3 lectureDL.py -v -all` will download all the lecture videos available. Does that also come under the heading of command-line arguments or is that something else?

**Comment by yoshw** [comment]
Yep, `-v -all` would be your command-line arguments in that case. As mentioned in the other issue, the `argparse` library is the best way to process command-line args in Python. Check out https://pymotw.com/3/argparse/

**Comment by larryhudson** [comment]
I've added a main() function as well as some command-line arguments. I chose to use the argv array to handle the arguments because it seemed more straightforward than argparse for my uses. Are there any reasons for me to use argparse over argv if I'm only handling simple requests? Thanks!

**Comment by yoshw** [comment]
Nice work. I think implementing your own argument-handling with `argv` is a reasonable approach for now, and a good learning experience. If you get to the point where the number and variety of arguments is more complex, or you want more flexibility or robustness, `argparse` is worth investigating.
